core/modules/search/search.module

```diff
@@ -975,13 +975,26 @@ function search_form($form, &$form_state, $action = '', $keys = '', $module = NU
  * @see search_box_form_submit()
  */
 function search_box($form, &$form_state, $form_id) {
+  $search_field_placeholder = config_get('search.settings', 'search_field_placeholder');
+  if (!empty($search_field_placeholder)) {
+    $attributes = array(
+      'title' => t('Enter the terms you wish to search for.'), 
+      'placeholder' => t($search_field_placeholder)
```

Comment:
Not sure how/when, but should this variable have to be sanitized somehow ?

Reply:
We assume access to this for trusted user only, anyway `t($search_field_placeholder)` must be changed, when we have done translatable configuration.